### PR TITLE
Fixing module startup behavior

### PR DESCRIPTION
**Pull request by mmarkwort**

### CODE REVIEW

**Comment by Toxantron** [comment]
The more I think about this, I am questioning the automatic restart in general. In the past 8 years I can't remember a valid case where a module fails to start and without any human interference could be restarted roughly 15ms later. Einstein once said:
> Insanity is doing the same thing over and over and expecting different results.
With that in mind I vote to remove that whole feature including configuration etc.

@dbeuchler and @slawor, any thoughts?

**Comment by dbeuchler** [comment]
Same for me, we started to remove this feature while
we decided to remove module stops in IParallelOperations.

**Comment by mmarkwort** [comment]
I created a new issue https://github.com/PHOENIXCONTACT/MORYX-Platform/issues/20 that addresses the automatic restart feature

File: src/Moryx.Runtime.Kernel/Modules/ComponentContracts/IModuleFailureStrategy.cs [hunk 1]
```diff
@@ -10,9 +10,5 @@ namespace Moryx.Runtime.Kernel
     /// </summary>
     internal interface IModuleFailureStrategy
```
Comment (Toxantron) [inline src/Moryx.Runtime.Kernel/Modules/ComponentContracts/IModuleFailureStrategy.cs:11]: I think the entire interface is obsolote now

File: src/Moryx.Runtime.Kernel/Modules/Components/FailureStrategy.cs [hunk 2]
```diff
@@ -1,9 +1,6 @@
 // Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 // Licensed under the Apache License, Version 2.0
 
-using Moryx.Modules;
-using Moryx.Runtime.Modules;
-
 namespace Moryx.Runtime.Kernel
```
Comment (Toxantron) [inline src/Moryx.Runtime.Kernel/Modules/Components/FailureStrategy.cs:4]: Same here

File: src/Moryx.Runtime.Kernel/Modules/ComponentContracts/IModuleFailureStrategy.cs (hunk 1 above)
Comment (mmarkwort) [reply src/Moryx.Runtime.Kernel/Modules/ComponentContracts/IModuleFailureStrategy.cs:11]: Ok, I will remove it.

Comment (Toxantron) [reply src/Moryx.Runtime.Kernel/Modules/ComponentContracts/IModuleFailureStrategy.cs:11]: Thanks

Comment (mmarkwort) [reply src/Moryx.Runtime.Kernel/Modules/ComponentContracts/IModuleFailureStrategy.cs]: Done

File: src/Moryx.Runtime.Kernel/Modules/Components/FailureStrategy.cs (hunk 2 above)
Comment (mmarkwort) [reply src/Moryx.Runtime.Kernel/Modules/Components/FailureStrategy.cs]: Done

File: src/Moryx.Runtime.Kernel/Modules/Components/ModuleStarter.cs [hunk 3]
```diff
@@ -62,7 +62,11 @@ private void StartModule(IServerModule module)
             }
             else
             {
-                ThreadPool.QueueUserWorkItem(ExecuteModuleStart, module);
+                // Don't try to start modules which initialization has been failed
+                if (module.State != ServerModuleState.Failure)
+                {
+                    ThreadPool.QueueUserWorkItem(ExecuteModuleStart, module);
```
Comment (Toxantron) [inline src/Moryx.Runtime.Kernel/Modules/Components/ModuleStarter.cs:68]: This seams redundant, but we will address it in #23.

**Comment by Toxantron** [comment]
Thank you for your contribution

**Pull request merged**